### PR TITLE
LoRA

### PR DESCRIPTION
**Pull request by lakshith-403**

### CODE REVIEW

File: docs/transformers/LoRA/GPT2.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+    def forward(self, x):
+        size_out = x.size()[:-1] + (self.nf,)
+        x = torch.addmm(self.bias, x.view(-1, x.size(-1)), self.weight)
+        x = x.view(size_out)
+        return x
+
+
+# from transformers
+class NewGELUActivation(nn.Module):
```
Comment (vpj) [inline docs/transformers/LoRA/GPT2.py:78]: does nn.GeLU produce different results?

File: docs/transformers/LoRA/GPT2.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    "use_cache": True,
+    "vocab_size": 50257
+}
+
+import math
+from torch import Tensor
+
+
+# from transformers
+class Conv1D(nn.Module):
```
Comment (vpj) [inline docs/transformers/LoRA/GPT2.py:52]: Use nn.Linear

File: docs/transformers/LoRA/GPT2.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        super().__init__()
+        self.embed_dim = config['n_embd']
+        self.num_heads = config['n_head']
+        self.head_dim = self.embed_dim // self.num_heads
+        self.split_size = self.embed_dim
+
+        self.c_att = Conv1D(config['n_embd'] * 3, config['n_embd'])
+        self.c_proj = Conv1D(config['n_embd'], config['n_embd'])
+
+        self.resid_dropout = nn.Dropout(config['resid_pdrop'])
```
Comment (vpj) [inline docs/transformers/LoRA/GPT2.py:115]: Lets remove dropout for simplicity

File: docs/transformers/LoRA/GPT2.py (hunk 1 above)
Comment (lakshith-403) [reply docs/transformers/LoRA/GPT2.py:78]: didnt know there was one. will check
